src/ApplicationDefaultCredentials.php

```diff
@@ -126,12 +155,38 @@ public static function getMiddleware(
     }
 
     /**
-     * Obtains an AuthTokenMiddleware which will fetch an access token to use in
-     * the Authorization header. The middleware is configured with the default
-     * FetchAuthTokenInterface implementation to use in this environment.
+     * Obtains an ProxyAuthTokenMiddleware that uses the default FetchAuthTokenInterface
+     * implementation to use in this environment.
      *
      * If supplied, $scope is used to in creating the credentials instance if
-     * this does not fallback to the Compute Engine defaults.
+     * this does not fallback to the compute engine defaults.
+     *
+     * @param string|array scope the scope of the access request, expressed
+     *        either as an Array or as a space-delimited String.
+     * @param callable $httpHandler callback which delivers psr7 request
+     * @param array $cacheConfig configuration for the cache when it's present
+     * @param CacheItemPoolInterface $cache A cache implementation, may be
+     *        provided if you have one already available for use.
+     * @param string $quotaProject specifies a project to bill for access
+     *   charges associated with the request.
+     * @return ProxyAuthTokenMiddleware
+     * @throws DomainException if no implementation can be obtained.
+     */
+    public static function getProxyMiddleware(
```

Comment:
Is there a use-case for sending an access token to `Proxy-Authentication`? If we don't know of one, I'd prefer to omit this for now for simplicity. Especially since, if someone does need it, they can easily create it manually:

```php
$creds = ApplicationDefaultCredentials::getCredentials($scope);
$middleware = new ProxyAuthTokenMiddleware($creds);
// etc...
```

Reply:
I am using this to send the ID token only, so I will remove it.